pkg/helm/ociClient.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2022 the Kubeapps contributors.
+// SPDX-License-Identifier: Apache-2.0
+
+package helm
+
+import (
+	"net/http"
+
+	"github.com/containerd/containerd/remotes"
+	"github.com/containerd/containerd/remotes/docker"
+	helmregistry "helm.sh/helm/v3/pkg/registry"
+)
+
+type (
+	// Derived from https://github.com/helm/helm/blob/v3.8.0/pkg/registry/client.go
+	IOCIClient interface {
+		Login(host string, options ...helmregistry.LoginOption) error
+		Logout(host string, opts ...helmregistry.LogoutOption) error
+		Pull(ref string, options ...helmregistry.PullOption) (*helmregistry.PullResult, error)
+		Push(data []byte, ref string, options ...helmregistry.PushOption) (*helmregistry.PushResult, error)
+		Tags(ref string) ([]string, error)
+	}
+
+	// Factory and wrapper for creating OCI clients (so that we an fake it for testing)
+	IOCIClientFactory interface {
+		BuildOCIClient(resolver remotes.Resolver) (IOCIClient, error)
+	}
+	OCIClientFactory        struct{}
+	OCIClientFactoryWrapper struct{ OCIClientFactory }
+)
+
+// GetResolver returns a containerd resolver configured for the given headers and http client
+func GetResolver(headers http.Header, netClient *http.Client) remotes.Resolver {
```

Comment:
The single point where we are creating a resolver from params.

Furthermore, I've read contained is deprecating the `Client: xxx` property in favor of `Hosts`, so I've already performed that change. However, ORAS's code (used by helm), still uses `Client`. I still have to perform a thorough IRL test no ensure we aren't adding any regressions here.